x-pack/plugins/security/public/components/breadcrumb.tsx

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { createContext, useEffect, useRef, useContext, FunctionComponent } from 'react';
+import { EuiBreadcrumb } from '@elastic/eui';
+import { useKibana } from '../../../../../src/plugins/kibana_react/public';
+
+interface BreadcrumbsContext {
+  parents: BreadcrumbProps[];
+  onMount(breadcrumbs: BreadcrumbProps[]): void;
+  onUnmount(breadcrumbs: BreadcrumbProps[]): void;
+}
+
+const BreadcrumbsContext = createContext<BreadcrumbsContext | undefined>(undefined);
+
+export interface BreadcrumbProps extends EuiBreadcrumb {
+  text: string;
+}
+
+/**
+ * Component that automatically sets breadcrumbs and document title based on the render tree.
+ *
+ * @example
+ * // Breadcrumbs will be set to: "Users > Create"
+ * // Document title will be set to: "Create - Users"
+ *
+ * ```typescript
+ * <Breadcrumb text="Users">
```

Comment:
Does this service also provide the ability to pass an `href` and/or `onClick`?

Reply:
Yes, the component extends the props from `EuiBreadcrumb` so anything that's valid there can also be passed into this component including `href`, `onClick` and `truncate`. (And whatever might be added in the future)

Reply:
Awesome thanks!